src/Observables.jl

```diff
@@ -64,12 +64,19 @@ function Base.getproperty(obs::Observable, field::Symbol)
 end
 
 """
-    notify!(observable::AbstractObservable)
+    notify(observable::AbstractObservable)
 
-Pushes an updates to all listeners of `observable`
+Update all listeners of `observable`.
 """
-function notify!(observable::AbstractObservable)
-    observable[] = observable[]
+function Base.notify(observable::AbstractObservable)
+    val = observable[]
+    for f in listeners(observable)
+        if f isa InternalFunction
+            f(val)
```

Comment:
I should point out that this bypass will need to go away (see #50), but I have that queued for a later pull request. (This is the second in a series of 9.)